karpenter-operator/controllers/nodeclass/ec2_nodeclass_controller.go

```diff
@@ -442,3 +445,23 @@ func (r *EC2NodeClassReconciler) mapToOpenShiftEC2NodeClasses(ctx context.Contex
 
 	return requests
 }
```

Comment:
I would remove setting the variable and assign the content directly to the spec:

```suggestion
	ec2NodeClass.Spec = awskarpenterv1.EC2NodeClassSpec{
		UserData:  ptr.To(string(userDataSecret.Data["value"])),
		AMIFamily: ptr.To("Custom"),
			},
		},
		AssociatePublicIPAddress: openshiftEC2NodeClass.Spec.AssociatePublicIPAddress,
		Tags:                     mergeEC2NodeClassTags(openshiftEC2NodeClass, hcp),
		DetailedMonitoring:       openshiftEC2NodeClass.Spec.DetailedMonitoring,
		BlockDeviceMappings:      openshiftEC2NodeClass.Spec.KarpenterBlockDeviceMapping(),
		InstanceStorePolicy:      openshiftEC2NodeClass.Spec.KarpenterInstanceStorePolicy(),

	return requests
Comment view}
}
```